go/runtime/bundle/manager.go

```diff
@@ -236,20 +248,45 @@ func (m *Manager) Queue(runtimeID common.Namespace, manifestHashes []hash.Hash)
 	}
 	m.downloadQueue[runtimeID] = hashes
 
-	// Trigger immediate download of new bundles.
+	// Trigger immediate download and clean-up of stale bundles.
 	select {
-	case m.downloadCh <- struct{}{}:
+	case m.downloadAndCleanCh <- struct{}{}:
+	default:
+	}
+}
+
+// Cleanup initiates the clean-up of regular bundles for the specified runtime,
+// removing versions lower than the specified one.
+func (m *Manager) Cleanup(runtimeID common.Namespace, active version.Version) {
+	// Update the active versions.
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	m.cleanupQueue[runtimeID] = active
+
+	// Trigger immediate download and clean-up of stale bundles.
+	select {
+	case m.downloadAndCleanCh <- struct{}{}:
 	default:
 	}
 }
 
 // Download tries to download bundles in the queue.
-func (m *Manager) Download() {
+func (m *Manager) download() {
 	for runtimeID := range m.runtimeIDs {
 		m.downloadBundles(runtimeID)
 	}
 }
 
+// Clean removes outdated manifest and deletes corresponding
+// exploded bundles for runtimes in the clean-up queue.
+func (m *Manager) Clean() {
+	m.logger.Info("removing regular bundles")
```

Comment:
Why not `cleaning bundles`, as you can immediately connect the log with the `Clean` function. The current messages  is not correct, as it says that all regular bundles will be removed.